### DISCUSSION TITLE
flux suspend and helm uninstall

### DISCUSSION THREAD

**Discussion by thecanadianroot** [fluxcd/flux2#2323] [Q&A]
Hi,

I've just encountered a behavior that I am not sure if it's desired. I'm asking here first rather than straight up opening an issue about it.

Is it normal that when we uninstall an Helm release (helm uninstall), a flux suspend on that same Helm release stays in place? Shouldn't the suspend be removed on helm uninstall?

It took me a while to find out why Flux wasn't doing anything with git changes to my new Helm release that used the same name as the one that got suspended then uninstalled.

Thanks

**Comment by kingdonb** [comment]
Helm and `helm uninstall` only interact with the Helm secret and do not know about the Flux HelmRelease resource.

Flux resources don't subscribe to those Helm secrets, when Flux reconciles a HelmRelease, it does a 3-way merge: checking the Helm secret, the HelmRelease inputs (any values and/or valuesFrom), and the helm chart for updates.

If you are not using the `reconcileStrategy` of `Revision` with a `GitRepository` source, then Helm Controller will not consider the Helm Chart to have changed unless the ChartVersion has been incremented. This is a quirk of the difference between how Helm considers its source immutable and the actual runtime capabilities of a Git repository, which can change between versions.

HelmRelease will not change based on updates made to the Helm Secret, they are separate inputs. If you use the Helm CLI to uninstall a Helm release and the HelmRelease is `suspend`ed, it will stay uninstalled until you `flux resume helmrelease` the HelmRelease.

Hope this helps! It sounds like you're describing expected behavior.
